framework/src/engine/engine.ts

```diff
@@ -104,6 +104,9 @@ export class Engine {
 	public async start() {
 		await this._init();
 		await this._network.start();
+		if (this._config.legacy.sync) {
+			await this._legacyChainHandler.syncBlocks();
```

Comment:
maybe add a log.info statement before after this call as the node will be busy with syncing for a while.